pkg/plugin/plugin.go

```diff
@@ -0,0 +1,57 @@
+package plugin
+
+import (
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/xeipuuv/gojsonschema"
+)
+
+// TODO bad name for this
+type PluginContainer struct {
+	Plugin Plugin
+	schema *gojsonschema.Schema
+}
+
+func NewPluginContainer(newPlugin NewFunc) (PluginContainer, error) {
+	var err error
+	pc := PluginContainer{}
+	pc.Plugin, err = newPlugin()
```

Comment:
This is putting stuff into the field of `pc` but then `pc` is returned later (below) in a half-constructed state. It is better to construct `PluginContainer{}` right at the place where it is returned to avoid that. May lead to "working" code in case an error check is missing.